docs/proposals/kubernetes-mcpgroup-crd.md

```diff
@@ -0,0 +1,272 @@
+kind: MCPGroup
+metadata:
+  name: engineering-team
+  namespace: default
+spec:
+  description: "Engineering team servers"
+---
+# Create servers in group
+apiVersion: mcp.toolhive.stacklok.io/v1alpha1
+kind: MCPServer
+metadata:
+  name: github-server
+spec:
+  image: ghcr.io/stackloklabs/github:latest
+  group: engineering-team
+---
+apiVersion: mcp.toolhive.stacklok.io/v1alpha1
+kind: MCPServer
+metadata:
+  name: jira-server
+spec:
+  image: ghcr.io/company/jira:latest
+  group: engineering-team
+```
+
+### Virtual MCP Usage
+
+```yaml
+# Virtual MCP references the group
+apiVersion: mcp.toolhive.stacklok.io/v1alpha1
+kind: MCPVirtualServer
+metadata:
+  name: engineering-virtual
+spec:
+  # References existing group
+  group: engineering-team
+
+  # Virtual MCP configuration
+  aggregation:
+    conflictResolution: prefix
+```
+
+### Querying Servers in Group
+
+```bash
+# List all servers in a group
+kubectl get mcpservers -n default --field-selector spec.group=engineering-team
+
+# Check group status
+kubectl get mcpgroup engineering-team -o jsonpath='{.status.servers}'
+```
+
+## Migration from CLI
+
+CLI groups and Kubernetes groups are separate concepts:
+- **CLI groups**: Local runtime groups (`.toolhive/` directory)
+- **K8s groups**: Cluster-scoped groups (etcd)
+
+No automatic migration - users manually create MCPGroup resources and set `spec.group` on MCPServers.
+
+## Type Definitions
+
+```go
+// MCPGroupSpec defines the desired state of MCPGroup
+type MCPGroupSpec struct {
+    // Description provides human-readable context
+    // +optional
+    Description string `json:"description,omitempty"`
+}
+
+// MCPGroupStatus defines observed state
+type MCPGroupStatus struct {
+    // Phase indicates current state
+    // +optional
+    Phase MCPGroupPhase `json:"phase,omitempty"`
+
+    // Servers lists server names in this group
+    // +optional
+    Servers []string `json:"servers,omitempty"`
+
+    // ServerCount is the number of servers
+    // +optional
+    ServerCount int `json:"serverCount"`
+
+    // Conditions represent observations
+    // +optional
+    Conditions []metav1.Condition `json:"conditions,omitempty"`
+}
+
+type MCPGroupPhase string
+
+const (
+    MCPGroupPhaseReady MCPGroupPhase = "Ready"
+)
+
+// Add to MCPServerSpec
+type MCPServerSpec struct {
+    // Existing fields...
+
+    // Group is the MCPGroup this server belongs to
+    // Must reference an existing MCPGroup in the same namespace
+    // +optional
+    Group string `json:"group,omitempty"`
+}
+```
+
+## Open Questions
+
+1. **Should group be immutable after creation?**
+   - Recommendation: Allow changes, easier for user corrections
+
+2. **What happens if group is deleted?**
+   - Recommendation: Servers continue running, `spec.group` becomes dangling reference
+   - Add webhook to warn on group deletion if servers reference it
+
+3. **Should we validate group exists on MCPServer create?**
+   - Recommendation: Yes, via validating webhook
```

Comment:
Not sure we need a webhook for this, seems a bit much, wouldn't error/warn logs in the Operator be enough? Perhaps with a status event for the `MCPServer`?